contracts/PoolManager.sol

```diff
@@ -251,11 +253,11 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, ERC1155, IERC1155Rec
             }
         }
 
+        PoolId id = key.toId();
         // Set the total swap fee, either through the hook or as the static fee set an initialization.
         uint24 totalSwapFee;
         if (key.fee.isDynamicFee()) {
-            totalSwapFee = IDynamicFeeManager(address(key.hooks)).getFee(msg.sender, key, params, hookData);
-            if (totalSwapFee >= 1000000) revert FeeTooLarge();
+            totalSwapFee = pools[id].slot0.dynamicFee;
```

Comment:
wondering how the optimizer handles fetching slot0 here vs fetching it in Pool.swap -- is it two sloads or does it somehow optimize one of them out? If it duplicates sloads I would consider how we could unify them to further benefit from the caching

Reply:
Maybe idea is that if we store both fees in slot0, we can just have Pool.swap(..) not even take fee as a swap param but read it from slot0 instead

Reply:
ah thats a good point. i can just return the swapFee from this func. alternatively we can remove it from event too. (did the first option but lmk what you think)